--- dolfin/la/EigenLUSolver.h.orig	2023-03-07 10:35:25.000000000 +0100
+++ dolfin/la/EigenLUSolver.h	2023-03-07 10:35:40.000000000 +0100
@@ -83,9 +83,10 @@
     std::string parameter_type() const
     { return "lu_solver"; }
 
+    class EigenLUImplBase;
+
   private:
     // Eigen LU implementation class
-    class EigenLUImplBase;
     std::unique_ptr<EigenLUImplBase> _impl;
 
     // Available LU solvers and descriptions
