--- Mongoose/Source/Mongoose_CSparse.cpp.old	2022-10-27 00:13:33.000000000 +0200
+++ Mongoose/Source/Mongoose_CSparse.cpp	2022-10-27 00:13:42.000000000 +0200
@@ -266,7 +266,7 @@
         return (NULL); /* out of memory */
     A->m     = m;      /* define dimensions and nzmax */
     A->n     = n;
-    A->nzmax = nzmax = std::max(nzmax, 1L);
+    A->nzmax = nzmax = std::max(nzmax, (csi) 1L);
     A->nz            = triplet ? 0 : -1; /* allocate triplet or comp.col */
     A->p             = (csi *)SuiteSparse_malloc(
         static_cast<size_t>(triplet ? nzmax : n + 1), sizeof(csi));
