--- sympy/core/basic.py	2023-03-07 07:39:47.000000000 +0100
+++ sympy/core/basic.py	2023-03-07 07:39:55.000000000 +0100
@@ -1,6 +1,7 @@
 """Base class for all the objects in SymPy"""
 from __future__ import print_function, division
-from collections import Mapping, defaultdict
+from collections import defaultdict
+from collections.abc import Mapping
 
 from .assumptions import BasicMeta, ManagedProperties
 from .cache import cacheit
--- sympy/assumptions/sathandlers.py	2023-03-07 07:41:33.000000000 +0100
+++ sympy/assumptions/sathandlers.py	2023-03-07 07:41:24.000000000 +0100
@@ -1,6 +1,7 @@
 from __future__ import print_function, division
 
-from collections import MutableMapping, defaultdict
+from collections import defaultdict
+from collections.abc import MutableMapping
 
 from sympy.core import (Add, Mul, Pow, Integer, Number, NumberSymbol,)
 from sympy.core.numbers import ImaginaryUnit
--- sympy/plotting/plot.py	2023-03-07 08:21:40.000000000 +0100
+++ sympy/plotting/plot.py	2023-03-07 08:21:47.000000000 +0100
@@ -25,7 +25,7 @@
 from __future__ import print_function, division
 
 import inspect
-from collections import Callable
+from collections.abc import Callable
 import warnings
 import sys
 
